x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -37,53 +41,113 @@ export const uniqueMonitorIds = (items: GetMonitorStatusResult[]): Set<string> =
     return acc;
   }, new Set<string>());
 
+const MESSAGE_AVAILABILITY_MAX = 3;
+
+export const availabilityMessage = (
+  availabilityResult: GetMonitorAvailabilityResult[],
+  threshold: string
+): string => {
+  return availabilityResult.length > 0
+    ? availabilityResult
+        .sort((a, b) => (a.availabilityRatio ?? 100) - (b.availabilityRatio ?? 100))
+        .slice(0, MESSAGE_AVAILABILITY_MAX)
+        .map(({ availabilityRatio, name, monitorId, url }) =>
+          i18n.translate('xpack.uptime.alerts.availability.monitorSummary', {
+            defaultMessage: '{nameOrId}({url}): {availabilityRatio}%',
+            values: {
+              nameOrId: name || monitorId,
+              url,
+              availabilityRatio: ((availabilityRatio ?? 1.0) * 100).toPrecision(5),
+            },
+          })
+        )
+        .reduce((prev: string, cur: string, _ind: number, array: string[]) => {
+          let next = '';
+          if (prev === '') {
```

Comment:
It'd be more clear to use `!==` to short circuit a return rather than double nesting the if.

Reply:
I try to get error conditions out of the way early if possible to reduce nesting when I can

Reply:
++ - thanks for pointing this out. I haven't been able to iterate on the code quality as much as I would like before opening this to review. https://github.com/elastic/kibana/pull/70284/commits/80c9983d469549c6087b28446d04e681bc2c26ac